tests/test_reader.py

```diff
@@ -261,3 +276,99 @@ def test_setup_none_in_path(self):
         DataReader(paths, framenames)
+
+
+
+class TestJsonLines:
+
+    def setUp(self):
+        self.filepaths = []
+
+        self.expected = pd.DataFrame({'id': ['001', '002', '003'],
+                                      'feature1': [1, 2, 3],
+                                      'feature2': [1.5, 2.5, 3.5]})
+
+
+    def tearDown(self):
+        for path in self.filepaths:
+            if os.path.exists(path):
+                os.unlink(path)
+        self.filepaths = []
+
+
+    @staticmethod
+    def create_jsonlines_file(jsondict):
+        tempf = tempfile.NamedTemporaryFile(mode='w',
+                                            suffix='.jsonlines',
+                                            delete=False)
+        for entry in jsondict:
+            json.dump(entry, tempf)
+            tempf.write('\n')
+        return tempf.name
+
+    def check_jsonlines_output(self, jsondict):
+        fname = self.create_jsonlines_file(jsondict)
+        self.filepaths.append(fname)
+        df = read_jsonlines(fname, converters={'id': str})
+        print(df)
+        assert_frame_equal(df.sort_index(axis=1), self.expected.sort_index(axis=1))
+
+
+    def test_read_jsonlines(self):
+        jsonlines = [{'id': '001',
+                      'feature1': 1,
+                      'feature2': 1.5},
+                     {'id': '002',
+                      'feature1': 2,
+                      'feature2': 2.5},
+                     {'id': '003',
+                      'feature1': 3,
+                      'feature2': 3.5}]
+        self.check_jsonlines_output(jsonlines)
+
+    def test_read_nested_jsonlines(self):
+        nested_jsonlines = [{'id': '001',
+                            'features': {'feature1': 1,
+                                         'feature2': 1.5}},
+                            {'id': '002',
+                            'features': {'feature1': 2,
+                                         'feature2': 2.5}},
+                            {'id': '003',
+                            'features': {'feature1': 3,
+                                         'feature2': 3.5}}]
+        self.check_jsonlines_output(nested_jsonlines)
+
+
+    def test_read_nested_jsonlines_all_nested(self):
+        all_nested_jsonlines = [{'values': {'id': '001',
+                                           'feature1': 1,
+                                           'feature2': 1.5}},
+                                {'values': {'id': '002',
+                                           'feature1': 2,
+                                           'feature2': 2.5}},
+                                {'values': {'id': '003',
+                                           'feature1': 3,
+                                           'feature2': 3.5}}]
+        self.check_jsonlines_output(all_nested_jsonlines)
+
+    def test_read_jsonlines_more_than_2_levels(self):
+        mlt_nested_jsonlines = [{'values': {'id': '001',
```

Comment:
```suggestion
        multi_nested_jsonlines = [{'values': {'id': '001',
```

Just a nitpick about variable naming.